### PR TITLE
Add new implementation

### PR DESCRIPTION
**Pull request by shredlocker**
# Why?

Add the implementation of the new algorithm using hash values

# What?

- Added new implementation
- Comments to describe the algorithm
- Documentation explaining what the algorithm is designed for

### CODE REVIEW

**Comment by bstien** [comment]
You should probably remove the `Item` struct, if that's not needed anymore

File: Diffuse/Shared/Diffuse.swift [hunk 1]
```diff
@@ -70,73 +70,83 @@ public struct Diffuse {
...
-                return $0.offset < $1.offset
+        // We can't remove more items than the amount of element in 'old' etc.
+        removed.reserveCapacity(old.count)
+        moved.reserveCapacity(minSize)
+        updated.reserveCapacity(minSize)
+
+        // 2 - Skip all the equal elements in the beginning of the arrays
+        // swiftlint:disable for_where
+        var startIndex = 0
+        for i in 0..<minSize {
```
Comment (ag12) [inline Diffuse/Shared/Diffuse.swift:104]: very minor stuff .. but we should use `0 ..< min` or without space.

**Comment by vadymmarkov** [comment]
I actually removed `Item` and some tests on https://github.com/finn-no/Diffuse/tree/ultimate-algorithm-fight branch.

**Comment by shredlocker** [comment]
I wasn't sure if you wanted me to make a PR with the that branch, because of the benchmark example. But we might want to have that as well?

**Comment by vadymmarkov** [comment]
The benchmark example is already on master, and I removed references to `diff2` from there. So it should be fine to make a PR with that branch.

**Comment by shredlocker** [comment]
I'll do that instead ;)

**Pull request closed without merging**